### DISCUSSION TITLE
[PIV]相关问题专楼

### DISCUSSION THREAD

**Discussion by cronfox** [canokeys/canokey-product#8] [Q&A]
那我就开个专楼吧((((
虽然我用不上 但之后肯定要搞(

**Comment by dangfan** [comment]
目前记录了两个问题，一个是DO小了，无法保存大证书，另一个是缺minidriver，还有别的问题吗

**Comment by mochaaP** [comment]
不兼容 piv-go 的 SetMetadata

<https://github.com/go-piv/piv-go/blob/48282da9879ffb328b7a100fe35ad7ba165fabde/piv/piv.go#L692>

**Reply by mochaaP** [reply]
未实现 Metadata

## Yubico Docs
<https://docs.yubico.com/yesdk/users-manual/application-piv/commands.html#get-metadata>
<https://docs.yubico.com/yesdk/users-manual/application-piv/apdu/metadata.html>

## canokey-core
<https://github.com/canokeys/canokey-core/blob/8218696bed430a370c80d5f40951666c2edd84a4/include/piv.h> 里没有实现 INS = 0xF7

![image](https://user-images.githubusercontent.com/21154023/148657386-3975f496-2709-4fd9-b55c-4baa0163e8a0.png)

cc @z4yx 下个版本是否考虑实现

**Reply by z4yx** [reply]
需要调研一下，这些元数据是否一定需要通过私有接口才能获得，如果能走标准接口的，尽量还是用标准接口吧。

**Reply by dangfan** [reply]
F7是私有命令

**Comment by z4yx** [comment]
相关问题已在 2.0.0 固件版本解决
